ui/src/js/common/components/SvgButton.js

```diff
@@ -0,0 +1,111 @@
+        </g>
+      </svg>;      
+    }
```

Comment:
please use a switch case statement instead of if.

Reply:
Instead of using the SVG code of every button inside of this component, I recommend externalizing all the buttons to there own SVG files and use the button type to load the SVG files. This allows removing all the if's, reducing the code, makes the code more readable and new buttons are easier to add.

Reply:
The Problem is, that you can not just load the svg code, it is what I tried first, you have to customize the code (strip namespace attributes from <svg>, remove inline styles from <path>). In general I would agree, but in this case creating external components doesn't seem to have a benefit for me in that case.

Reply:
i think i found a solution to make the code more readable, please have a look at the last commit

i would prefer generating the html directly in the render method to avoid using **dangerouslySetInnerHTML={{ __html: svgButton }}.** 
but you get "unexpected token" errors when you push elements with a closing tag(e.g. **svgButton.push(</g>);**) which is required when creating these buttons.
please let me know if you have any ideas

Reply:
@Charokee what I meant is that the svg lays in its own file and we render an img tag that points to the svg file.

Reply:
You can load svgs with an img tag, but then the css fill property used for the hover color won't works as it requires inlining the svg:
https://css-tricks.com/using-svg/#article-header-id-6